src/main/java/net/minecraftforge/fml/packs/DelegatingResourcePack.java

```diff
@@ -84,69 +111,86 @@ public String getName()
     @Override
     public Collection<ResourceLocation> getAllResourceLocations(ResourcePackType type, String pathIn, String pathIn2, int maxDepth, Predicate<String> filter)
     {
-        synchronized(delegates)
-        {
-            return delegates.stream()
-                    .flatMap(r -> r.getAllResourceLocations(type, pathIn, pathIn2, maxDepth, filter).stream())
-                    .collect(Collectors.toList());
-        }
+        return delegates.stream()
+                .flatMap(r -> r.getAllResourceLocations(type, pathIn, pathIn2, maxDepth, filter).stream())
+                .collect(Collectors.toList());
     }
 
     @Override
     public Set<String> getResourceNamespaces(ResourcePackType type)
     {
-        synchronized (delegates)
-        {
-            return delegates.stream()
-                    .flatMap(r -> r.getResourceNamespaces(type).stream())
-                    .collect(Collectors.toSet());
-        }
+        return delegates.stream()
+                .flatMap(r -> r.getResourceNamespaces(type).stream())
+                .collect(Collectors.toSet());
```

Comment:
Could this be changed to use `namespaceAssets.keySet()`/`namespaceData.keyset()`? IIRC `ModFileResourcePack#getResourceNamespaces` does file system operations, so it's relatively slow.

Reply:
You're right, changed that.